implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -258,6 +258,34 @@ impl IdentitiesState {
         Ok(IdentityState { path, config })
     }
 
+    pub fn list(&self) -> Result<Vec<IdentityState>> {
+        let mut identities = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            let filename = entry.file_name().into_string()
+                .map_err(|_| CliStateError::Io(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                "identity has an invalid name",
+            )))?;
+            if !entry.file_type()?.is_dir() && filename.ends_with(".json") {
+                let identity_name = &filename[0..filename.len()-5];
+                identities.push(self.get(identity_name)?);
+            }
```

Comment:
Since all the necessary checks are done inside `self.get()`, this could be simplified to:

```rust
let identity_name = file_stem(&entry.path())?;
identities.push(self.get(&identity_name)?);
```